packages/perennial-oracle/contracts/ChainlinkFeedOracle.sol

```diff
@@ -0,0 +1,157 @@
+                    uint128(_aggregatorRoundIdToProxyRoundId(_latestPhaseId() + 1, 1))
+                )
+            );
+        }
+
+        // Return packaged oracle version
+        return _buildOracleVersion(round);
+    }
+
+    /**
+     * @notice Returns the current oracle version
+     * @return oracleVersion Current oracle version
+     */
+    function currentVersion() public view returns (OracleVersion memory oracleVersion) {
+        return _buildOracleVersion(aggregator.getLatestRound());
+    }
+
+    /**
+     * @notice Returns the current oracle version
+     * @param version The version of which to lookup
+     * @return oracleVersion Oracle version at version `version`
+     */
+    function atVersion(uint256 version) public view returns (OracleVersion memory oracleVersion) {
+        return _buildOracleVersion(aggregator.getRound(_versionToRoundId(version)), version);
+    }
+
+    /**
+     * @notice Builds an oracle version object from a Chainlink round object
+     * @dev Computes the version for the round
+     * @param round Chainlink round to build from
+     * @return Built oracle version
+     */
+    function _buildOracleVersion(ChainlinkRound memory round) private view returns (OracleVersion memory) {
+        Phase memory phase = _phases[round.phaseId()];
+        uint256 version = uint256(phase.startingVersion) + round.roundId - uint256(phase.startingRoundId);
+        return _buildOracleVersion(round, version);
+    }
+
+    /**
+     * @notice Builds an oracle version object from a Chainlink round object
+     * @param round Chainlink round to build from
+     * @param version Determined version for the round
+     * @return Built oracle version
+     */
+    function _buildOracleVersion(ChainlinkRound memory round, uint256 version)
+    private view returns (OracleVersion memory) {
+        Fixed18 price = Fixed18Lib.ratio(round.answer, _decimalOffset);
+        return OracleVersion({ version: version, timestamp: round.timestamp, price: price });
+    }
+
+    /**
+     * @notice Computes the chainlink round ID from a version
+     * @param version Version to compute from
+     * @return Chainlink round ID
+     */
+    function _versionToRoundId(uint256 version) private view returns (uint256) {
+        Phase memory phase = _versionToPhase(version);
+        return uint256(phase.startingRoundId) + version - uint256(phase.startingVersion);
+    }
+
+    /**
+     * @notice Computes the chainlink phase ID from a version
+     * @param version Version to compute from
+     * @return phase Chainlink phase
+     */
+    function _versionToPhase(uint256 version) private view returns (Phase memory phase) {
+        uint256 phaseId = _latestPhaseId();
+        phase = _phases[phaseId];
+        while (uint256(phase.startingVersion) > version) {
+            phaseId--;
+            phase = _phases[phaseId];
+            if (phaseId < _firstPhaseId) revert InvalidVersionRequested(version);
```

Comment:
this is redundant, should fail correctly with the original code as long as the phase map is populated correctly